### PR TITLE
Fix: Upgrade Failing due to bg capsule sync task

### LINKED ISSUE #207: Upgrade randomly fails in foreman-maintain
Upgrade jobs started randomly failing when we started with foreman-mainatin. This is because,

We sync, publish,promote repos in Lifecycle env. This in-turn starts Capsule Sync task which runs with delayed executioner.
We don't wait for the Sync task to complete and start with upgrade with foreman-maintain.Where foreman-maintain fails the job, if any task are in running state.

### PR DESCRIPTION
**Pull request by jyejare**
Fixes #207.

### CODE REVIEW

**Comment by jyejare** [comment]
This will also help us to merge Satellite 6.4 branch for upgrade as stated in issue #210  with milestone set to 31st Aug.

File: upgrade/helpers/tasks.py [hunk 1]
```diff
@@ -727,3 +728,46 @@ def puppet_autosign_hosts(version, hosts, append=True):
         'ver2': '/etc/puppetlabs/puppet/autosign.conf'}
     for host in hosts:
         run('echo "{0}" {1} {2}'.format(host, append, puppetfile[puppetver]))
+
+
+def get_satellite_host():
```
Comment (san7ket) [inline upgrade/helpers/tasks.py]: We already have this, somewhere . why duplicate

Comment (san7ket) [reply upgrade/helpers/tasks.py]: https://github.com/SatelliteQE/satellite6-upgrade/blob/63ddfad7563ffba4c862458a3b5dc11145f65545/upgrade_tests/helpers/scenarios.py#L132

Comment (jyejare) [reply upgrade/helpers/tasks.py]: I want to remove it from there and keep it here.

Comment (jyejare) [reply upgrade/helpers/tasks.py]: I changes that everywhere

**Pull request merged**